packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -171,7 +171,7 @@ export const useGridKeyboardNavigation = (
 
   const getRowIdFromIndex = React.useCallback(
     (rowIndex: number) => {
-      return currentPageRows?.[rowIndex].id;
+      return apiRef.current.getRowId(currentPageRows[rowIndex]);
```

Comment:
@mui/xgrid Just to formalize it: `row.id` is always wrong, we need `apiRef.current.getRowId` to properly support the `getRowId` prop.

Reply:
Isn't the `row` here a `GridRowEntry` where `id` is the correct id that takes `getRowId` into account ?

Reply:
Yes, accessing `GridRowEntry['id']` is OK:
https://github.com/mui/mui-x/blob/22e1c8ded97b789e8ed0b77b616b192b5c782da7/packages/grid/x-data-grid/src/models/gridRows.ts#L187-L196

Reply:
For me the `apiRef.current.getRowId` that uses the `getRowId` is a mistake.
It should use the lookup from the state.

Reply:
Right, my bad, thought I had a `GridRow`.